File: typescript-client-sdk/src/sdk/models/operations/getallcrmcontacts.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAllCrmContactsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;
}


export class GetAllCrmContactsResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contacts", elemType: shared.Contact })
  contacts?: shared.Contact[];

  @SpeakeasyMetadata({ data: "json, name=nextPageCursor" })
  nextPageCursor?: string;
}


export class GetAllCrmContactsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetAllCrmContactsQueryParams;
}


export class GetAllCrmContactsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetAllCrmContactsResponseBody;
}

File: go-client-sdk/pkg/models/operations/getallcrmcontacts.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetAllCrmContactsQueryParams struct {
	AccessToken string   `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool    `queryParam:"style=form,explode=true,name=allFields"`
	Cursor      *string  `queryParam:"style=form,explode=true,name=cursor"`
	Limit       *float64 `queryParam:"style=form,explode=true,name=limit"`
}

type GetAllCrmContactsResponseBody struct {
	Contacts       []shared.Contact `json:"contacts,omitempty"`
	NextPageCursor *string          `json:"nextPageCursor,omitempty"`
}

type GetAllCrmContactsRequest struct {
	QueryParams GetAllCrmContactsQueryParams
}

type GetAllCrmContactsResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetAllCrmContactsResponseBody
}
